write-rustdoc-hide-lines/src/formatter.rs

```diff
@@ -0,0 +1,84 @@
+use std::{
+    fs::{self, File},
+    io::{self, BufRead},
+    path::Path,
+};
+
+use crate::{
+    code_block_definition::CodeBlockDefinition, hidden_ranges::get_hidden_ranges,
+    utils::visit_dir_md_files,
+};
+
+pub fn run(dir: &Path) {
+    println!("Formatting folder: {:?}", dir);
+
+    let result = visit_dir_md_files(dir, &|entry| {
+        println!("{:?}", entry.path());
+
+        let file = File::open(entry.path())?;
+        let file_len = file.metadata().unwrap().len().try_into().unwrap();
+        let mut contents = String::with_capacity(file_len);
+        let mut is_inside_rust_code_block = false;
+        let mut rust_block: Vec<String> = vec![];
+
+        for line in io::BufReader::new(file).lines() {
+            let line = line?;
+            let is_code_block_open = line.starts_with("```rust");
+            let is_code_block_close = line == "```";
+
+            if is_inside_rust_code_block && is_code_block_open {
+                panic!("Nested '```rust' code block not allowed");
+            } else if is_code_block_open {
+                is_inside_rust_code_block = true;
+            }
+
+            // Skip the line, save it as is
+            if !is_inside_rust_code_block {
+                contents.push_str(&format!("{}\n", &line));
```

Comment:
According to clippy this should avoid allocating
```suggestion
				writeln!(&mut contents, "{}", &line)?;
```

Reply:
Oh, I just realized I was on an older Rust version. These weren't reported in `1.60`. I'll try to fix this, although I'm not sure what to do with the error type, as `writeln!` is not compatible with `std::io::Error`… I suppose I need to change `visit_dir_md_files` to `dyn Error` or something like that? (like when `main` returns `Result`)

Reply:
Ah, right, I added `anyhow` on my branch for that

Reply:
I've used `anyhow` in the end… and, I've found a couple of articles/videos on error handling that I'll check later. ^_^